src/vm/custommarshalerinfo.cpp

```diff
@@ -134,11 +134,14 @@ CustomMarshalerInfo::~CustomMarshalerInfo()
 {
     WRAPPER_NO_CONTRACT;
 #ifndef CROSSGEN_COMPILE    
-    if (m_hndCustomMarshaler)
+    if (m_pLoaderAllocator->IsAlive() && m_hndCustomMarshaler)
```

Comment:
Is this ever needed - can we depend on the loader allocator always taking care of freeing the handle?

Reply:
There’s a race when creating the CustomMarshalerInfo on different threads, and we destroy the extra CMInfo if another thread creates it before we do. That’s the only path that needs this destruction logic.